opentelemetry-exporter-geneva/geneva-uploader/src/payload_encoder/bond_encoder.rs

```diff
@@ -0,0 +1,338 @@
+
+    // Field ID
+    writer.write_all(&field.field_id.to_le_bytes())?;
+
+    // Type
+    writer.write_all(&[field.type_id])?;
+
+    // Additional type info (all zeros for primitives)
+    writer.write_all(&0u16.to_le_bytes())?; // struct_def
+    writer.write_all(&[0u8])?; // element
+    writer.write_all(&[0u8])?; // key
+    writer.write_all(&[0u8])?; // bonded_type
+    writer.write_all(&[0u8])?; // default_value_present
+
+    // Add 8 bytes padding after each field except the last one
+    if !is_last {
+        writer.write_all(&[0u8; 8])?;
+    }
+
+    Ok(())
+}
+
+/// Encode a payload with dynamic fields
+#[allow(dead_code)] // May be used in future
+pub(crate) fn encode_dynamic_payload<W: Write>(
+    writer: &mut W,
+    fields: &[FieldDef],
+    values: &[(&str, Vec<u8>)], // field_name -> encoded value
+) -> Result<()> {
+    // Write Simple Binary header
+    writer.write_all(&[0x53, 0x50])?; // 'S','P'
+    writer.write_all(&[0x01, 0x00])?; // Version 1
+
+    // Create a map for quick lookup
+    let value_map: std::collections::HashMap<&str, &[u8]> =
+        values.iter().map(|(k, v)| (*k, v.as_slice())).collect();
+
+    // Write values in field order
+    for field in fields {
+        if let Some(value_bytes) = value_map.get(field.name.as_str()) {
+            writer.write_all(value_bytes)?;
+        } else {
+            // Write default value based on type
+            match field.type_id {
+                7 => writer.write_all(&0f32.to_le_bytes())?, // float
+                8 => writer.write_all(&0f64.to_le_bytes())?, // double
+                9 | 18 => writer.write_all(&0u32.to_le_bytes())?, // empty string
+                16 => writer.write_all(&0i32.to_le_bytes())?, // int32
+                17 => writer.write_all(&0i64.to_le_bytes())?, // int64
+                _ => {}                                      // Handle other types as needed
+            }
+        }
+    }
+
+    Ok(())
+}
+
+pub(crate) struct BondEncodedSchema {
+    schema: DynamicSchema,
+    encoded_bytes: Vec<u8>,
+}
+
+impl BondEncodedSchema {
+    pub(crate) fn from_fields(
+        fields: &[(&str, u8, u16)],
+        struct_name: &str,
+        namespace: &str,
+    ) -> Self {
+        let mut schema = DynamicSchema::new(struct_name, namespace); //"OtlpLogRecord", "telemetry");
+
+        for (name, type_id, field_id) in fields {
+            schema.add_field(name, *type_id, *field_id);
+        }
+
+        let encoded_bytes = schema.encode().expect("Schema encoding failed");
+
+        Self {
+            schema,
+            encoded_bytes,
+        }
+    }
+
+    pub(crate) fn as_bytes(&self) -> &[u8] {
+        &self.encoded_bytes
+    }
+}
+
+impl Clone for BondEncodedSchema {
+    fn clone(&self) -> Self {
+        Self {
+            schema: self.schema.clone(),
+            encoded_bytes: self.encoded_bytes.clone(),
+        }
+    }
+}
+
+// Replacement for EncoderRow
+pub(crate) struct BondEncodedRow {
```

Comment:
Do we need this struct and its methods? It only seems to be cloning a vec of bytes.